src/ui/components/search/filtersearchcomponent.js

```diff
@@ -188,15 +198,18 @@ export default class FilterSearchComponent extends Component {
 
   /**
    * Perform the vertical search with all saved filters and query,
-   * optionally redirecting based on config
+   * optionally redirecting based on config. Uses window.setTimeout to allow
+   * other filters to finish rendering before searching.
    */
   search () {
     if (this._storeOnChange) {
       return;
     }
-    this.core.verticalSearch(this._config.verticalKey, {
-      resetPagination: true,
-      useFacets: true
+    window.setTimeout(() => {
```

Comment:
Why do we need setTimeout here? I'm not sure what the significance of letting other filters render first is

Reply:
Even before this PR, FilterBox onMount would call its _saveFiltersToStorage(), which would call core.setFilter. I noticed during testing that the on page load search ran by FilterSearch was missing filters that the on page load search ran by the SearchBar had. I believe this was because the onCreate of FilterSearch was finishing before the onMount of FilterBox. If I move my FilterSearch's addComponent call anywhere after my FilterBox's addComponent call of the ANSWERS.init onReady() this issue no longer occurs.

Reply:
I think I remember you being the one who told me about this originally but https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful